src/chain.cpp

```diff
@@ -58,6 +58,13 @@ const CBlockIndex *CChain::FindFork(const CBlockIndex *pindex) const {
     return pindex;
 }
 
+CBlockIndex *CChain::FindLatestBefore(int64_t nTime) const {
+    CBlockIndex *pindex = Tip();
+    while (pindex && pindex->pprev && pindex->GetBlockTime() > nTime)
```

Comment:
Agreed with @promag, a system can add the safety net of ~144 blocks to the timestamp if necessary and a person likely knows what they are doing if they opt in for this.